### PR TITLE
Libsed datastore

### PR DESCRIPTION
**Pull request by AndrzejJakowski**

### CODE REVIEW

File: src/lib/nvme_pt_ioctl.c [hunk 1]
```diff
@@ -1566,7 +1566,7 @@ static struct opal_req_item opal_read_datastr_cmd[] = {
 static int opal_read_datastr(int fd, struct opal_device *dev, uint8_t *data, uint64_t offset, uint64_t size)
 {
 	int ret = 0;
-	uint64_t len = 0, index = 0;
+	uint64_t len = 0, index = 0, end_row;
```
Comment (RevanthRajashekar) [inline src/lib/nvme_pt_ioctl.c]: Declare and initialize end_row = size - 1;

File: src/lib/nvme_pt_ioctl.c [hunk 2]
```diff
@@ -1588,10 +1588,11 @@ static int opal_read_datastr(int fd, struct opal_device *dev, uint8_t *data, uin
 		return -EINVAL;
 	}
 
-	while (index < size) {
+	end_row = size - 1;
```
Comment (RevanthRajashekar) [inline src/lib/nvme_pt_ioctl.c]: This line can be removed.

**Comment by RevanthRajashekar** [comment]
The patch description needs to be fixed/detailed.
Also, this patch set needs to be re-based because of merge conflict.

File: src/lib/nvme_pt_ioctl.c [hunk 3]
```diff
@@ -2039,53 +2064,29 @@ int opal_eraselr_pt(struct sed_device *dev, const char *password,
...
-	struct sed_key disk_key;
+	int ret = 0, opal_auth;
 	struct opal_device *opal_dev;
 
-	if (key == NULL || from == NULL) {
-		SEDCLI_DEBUG_MSG("Must provide the password and a valid source pointer\n");
+	if (to == NULL) {
+		SEDCLI_DEBUG_MSG("Must provide a valid destination pointer\n");
 		return -EINVAL;
 	}
```
Comment (RevanthRajashekar) [inline src/lib/nvme_pt_ioctl.c:2095]: One more check needs to be included here:
if (auth != SED_ANYBODY && key == NULL)
PRINT "User must provide a valid key"

File: src/lib/nvme_pt_ioctl.c (hunk 1 above)
Comment (AndrzejJakowski) [reply src/lib/nvme_pt_ioctl.c]: fixed

File: src/lib/nvme_pt_ioctl.c (hunk 2 above)
Comment (AndrzejJakowski) [reply src/lib/nvme_pt_ioctl.c]: done

**Comment by AndrzejJakowski** [comment]
> The patch description needs to be fixed/detailed.
> Also, this patch set needs to be re-based because of merge conflict.

done

File: src/lib/nvme_pt_ioctl.c (hunk 3 above)
Comment (AndrzejJakowski) [reply src/lib/nvme_pt_ioctl.c:2095]: this check has been added to sed.c

File: src/lib/sed.c [hunk 4]
```diff
@@ -298,46 +290,42 @@ int sed_eraselr(struct sed_device *dev, const char *password,
...
-	return curr_if->ds_admin_write_fn(dev, key, key_len, from, size, offset);
-}
-
-int sed_ds_admin_read(struct sed_device *dev, const char *key, uint8_t key_len,
-		uint8_t *to, uint32_t size, uint32_t offset)
-{
-	if (curr_if->ds_admin_read_fn == NULL)
-		return -EOPNOTSUPP;
+	if (auth != SED_ANYBODY && key == NULL) {
+		SEDCLI_DEBUG_MSG("Key can't be null\n");
```
Comment (RevanthRajashekar) [inline src/lib/sed.c:301]: How about, "Key can't be NULL for this authority"..? Just to be more clear.
But this is fine as well

File: src/lib/sed.c [hunk 5]
```diff
@@ -298,46 +290,42 @@ int sed_eraselr(struct sed_device *dev, const char *password,
...
 
-	return curr_if->ds_anybody_read_fn(dev, to, size, offset);
-}
-
-int sed_ds_anybody_write(struct sed_device *dev, uint8_t *from, uint32_t size, uint32_t offset)
-{
-	if (curr_if->ds_anybody_write_fn == NULL)
-		return -EOPNOTSUPP;
+	if (auth != SED_ANYBODY && key == NULL) {
+		SEDCLI_DEBUG_MSG("Key can't be null\n");
```
Comment (RevanthRajashekar) [inline src/lib/sed.c:316]: How about, "Key can't be NULL for this authority"..? Just to be more clear.
But this is fine as well

**Review by RevanthRajashekar — APPROVED** [review]

**Pull request merged**